store/rootmulti/store.go

```diff
@@ -594,18 +604,10 @@ func (rs *Store) Query(req abci.RequestQuery) abci.ResponseQuery {
 		return sdkerrors.QueryResult(sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "proof is unexpectedly empty; ensure height has not been pruned"))
 	}
 
-	// If the request's height is the latest height we've committed, then utilize
-	// the store's lastCommitInfo as this commit info may not be flushed to disk.
-	// Otherwise, we query for the commit info from disk.
-	var commitInfo *types.CommitInfo
 
```

Comment:
We now flush to disk as soon as `rs.lastCommitInfo` is updated. Although I acknowledge that db read is more expensive, I think it's safer to make a database read and let the DB handle synchronization. Also, this change makes it more readable imo